MaxText/layers/attentions.py

```diff
@@ -709,4 +737,127 @@ def __call__(self,
 
     # apply output projection,  output dim is set to the input dim.
     out = self.out_projection(inputs_q.shape[-1], out)
-    return out
\ No newline at end of file
+    return out
+
+
+class Gpt3MultiHeadAttention(nn.Module):
```

Comment:
Why is this different form the other implementations? It seems duplicative...

(Also if it really a gpt3 specific file, please put it in gpt3.py.)

Reply:
Gotcha.

Reply:
Have moved the class into `gpt3.py`

Reply:
Discussed offline and keep the `Gpt3MultiHeadAttention`.